### PR TITLE
Make unlabeled data hook work over tasks

### PR DESCRIPTION
**Pull request by supersoob**
This PR is to make unlabeled data hook work over tasks in semi-supervised learning.

### CODE REVIEW

**Review by jaegukhyun — APPROVED** [review]
LGTM

File: mpa/modules/hooks/unlabeled_data_hook.py [hunk 1]
```diff
@@ -2,15 +2,14 @@
...
 from mmcv.runner import get_dist_info
 from mmcv.runner import HOOKS, Hook
-
-from mmdet.datasets import build_dataset, build_dataloader
-
 from mpa.modules.datasets.composed_dataloader import ComposedDL
 from mpa.utils.logger import get_logger
 
 logger = get_logger()
+task_lib_name = dict(classification="mmcls", detection="mmdet", segmentation="mmseg")
```
Comment (goodsong81) [inline mpa/modules/hooks/unlabeled_data_hook.py]: Why don't you use capital letters for this kind of global & constant table?

File: mpa/modules/hooks/unlabeled_data_hook.py [hunk 2]
```diff
@@ -21,15 +20,22 @@ def __init__(
...
         seed=None,
         **kwargs
     ):
         super().__init__(**kwargs)
+
         # Build unlabeled dataset & loader
-        logger.info('In UnlabeledDataHook.before_epoch, creating unlabeled dataset...')
+        m = importlib.import_module(f"{task_lib_name[model_task]}.datasets")
+        build_dataset = getattr(m, "build_dataset")
+        build_dataloader = getattr(m, "build_dataloader")
```
Comment (goodsong81) [inline mpa/modules/hooks/unlabeled_data_hook.py]: m -> task_lib_module?

**Review by goodsong81 — COMMENTED** [review]
Generally looks OK.
You may want to revise some variable names for other developers :)

File: mpa/modules/hooks/unlabeled_data_hook.py (hunk 2 above)
Comment (supersoob) [reply mpa/modules/hooks/unlabeled_data_hook.py]: I fixed.

**Review by JihwanEom — APPROVED** [review]

**Pull request merged**